src/App.tsx

```diff
@@ -1,26 +1,114 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useMemo, useState, useRef } from 'react';
+import { UserWarning } from './utils/UserWarning';
+import { USER_ID, getTodos, deleteTodo } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Header } from './components/Header';
+import { FilterBy } from './types/FilterBy';
+import { Todo } from './types/Todo';
+import { allCompletedTodos } from './utils/Utils';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.All);
+  const [error, setError] = useState<string | null>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [processingTodos, setProcessingTodos] = useState<number[]>([]);
+  const fieldTitle = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => setError('Unable to load todos'));
+  }, []);
+
+  const getFilteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (filterBy) {
+        case FilterBy.Active:
+          return !todo.completed;
+        case FilterBy.Completed:
+          return todo.completed;
+        default:
+          return true;
+      }
+    });
+  }, [filterBy, todos]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const onAddTodo = (todo: Todo) => {
```

Comment:
with the prefix `on` we call just props
so better rename to `addTodo`
and finally `onAdd={addTodo}`